pyccel/ast/operators.py

```diff
@@ -1070,7 +1070,80 @@ class PyccelIsNot(PyccelIs):
     def __repr__(self):
         return '{} is not {}'.format(self.args[0], self.args[1])
 
+
 #==============================================================================
 
+class IfTernaryOperator(Basic, PyccelOperator):
+    """Represent a ternary conditional operator in the code, of the form (a if cond else b)
+
+    Parameters
+    ----------
+    args :
+        args : type list
+        format : condition , value_if_true, value_if_false
+
+    Examples
+    --------
+    >>> from sympy import Symbol
+    >>> from pyccel.ast.core import Assign
+	>>>	from pyccel.ast.operators import IfTernaryOperator
+    >>> n = Symbol('n')
+    >>> x = 5 if n > 1 else 2
+    >>> IfTernaryOperator(PyccelGt(n > 1),  5,  2)
+    IfTernaryOperator(PyccelGt(n > 1),  5,  2)
+    """
+    _precedence = 3
+    def __init__(self, cond, value_true, value_false):
```

Comment:
I agree with codacy. Superclass `__init__` methods should ideally always be called. `PyccelOperator.__init__` ensures that the dtype/precision/etc are set, however it does this by calling additional methods so if this Operator needs to implement it's own version of those it can do so by overriding the virtual methods

Reply:
It is also `PyccelOperator.__init__` which calls the functions which ensure that the parentheses are added where necessary so the bug will persist as long as it isn't called